extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/AuthenticationProviderService.java

```diff
@@ -205,38 +214,80 @@ private LDAPConnection bindAs(Credentials credentials)
      *     denied.
      */
     public AuthenticatedUser authenticateUser(Credentials credentials)
-            throws GuacamoleException {
-
-        // Attempt bind
-        LDAPConnection ldapConnection;
-        try {
-            ldapConnection = bindAs(credentials);
-        }
-        catch (GuacamoleException e) {
-            logger.error("Cannot bind with LDAP server: {}", e.getMessage());
-            logger.debug("Error binding with LDAP server.", e);
-            ldapConnection = null;
-        }
-
-        // If bind fails, permission to login is denied
-        if (ldapConnection == null)
-            throw new GuacamoleInvalidCredentialsException("Permission denied.", CredentialsInfo.USERNAME_PASSWORD);
-
-        try {
-
-            // Return AuthenticatedUser if bind succeeds
-            AuthenticatedUser authenticatedUser = authenticatedUserProvider.get();
-            authenticatedUser.init(credentials);
-            return authenticatedUser;
-
-        }
-
-        // Always disconnect
-        finally {
-            ldapService.disconnect(ldapConnection);
-        }
-
+    		throws GuacamoleException {
+
+    	// Attempt bind
+    	LDAPConnection ldapConnection;
+    	try {
+    		ldapConnection = bindAs(credentials);
+    	}
+    	catch (GuacamoleException e) {
+    		logger.error("Cannot bind with LDAP server: {}", e.getMessage());
+    		logger.debug("Error binding with LDAP server.", e);
+    		ldapConnection = null;
+    	}
+
+    	// If bind fails, permission to login is denied
+    	if (ldapConnection == null)
+    		throw new GuacamoleInvalidCredentialsException("Permission denied.", CredentialsInfo.USERNAME_PASSWORD);
+
+    	boolean authenticated=true;
```

Comment:
> only way will be ...

I don't think that *has* to be the only other possibility. There really must be a better way.

> however that will mean we have to repeat following code ...

If only there were a programming construct which allowed us to reuse pieces of code ... ;)

Anyway, it's clear the logic here needs some cleaning, but it's hard to read through given the other comments already made.

Perhaps we should revisit this after the stylistic issues are addressed?